src/Middleware/Spa/SpaProxy/src/SpaProxyLaunchManager.cs

```diff
@@ -25,10 +21,12 @@ internal class SpaProxyLaunchManager : IDisposable
 
         public SpaProxyLaunchManager(
             ILogger<SpaProxyLaunchManager> logger,
+            IHostApplicationLifetime appLifetime,
             IOptions<SpaDevelopmentServerOptions> options)
         {
             _options = options.Value;
             _logger = logger;
+            appLifetime.ApplicationStopping.Register(() => Dispose(true));
```

Comment:
So SpaProxyLaunchManager is a service in DI but Dispose isn't getting called on app shutdown for some reason?

Who normally calls StopAsync?

Do you want to use ApplicationStopped instead? Requests may still be in flight otherwise.

Reply:
@Tratcher the problem here is that the debugger kills the app and leaves the process running.

We don't get a chance to kill the SPA process ourselves. This change just brings the current implementation more inline with the old implementation, although I suspect both suffer from the same issue.

Reply:
If I recall correctly, the debugger kills processes in the rudest way possible with no chance to run shutdown logic. What you really need is a way to register the child process with the debugger as something it should also terminate. Maybe if you launched the child process with the debugger attached?

https://docs.microsoft.com/en-us/visualstudio/debugger/debug-multiple-processes?view=vs-2019#stop-debugging-with-multiple-processes

I see some prior art here:
https://github.com/dotnet/aspnetcore/issues/5204
https://github.com/dotnet/aspnetcore/issues/11597
https://github.com/dotnet/aspnetcore/pull/17883 (claims ApplicationStopping worked for them)

Reply:
None of this things work, people thought they worked because they didn't see the window (or didn't try with the debugger).

I found a reasonable solution which is simple enough I'm happy with. We launch a powershell process that monitors the .NET process and when it sees it die, it kills the npm process and all its children, and that does the trick.